tests/test_service.py

```diff
@@ -75,6 +75,47 @@ def test_index(self):
         self.assertEqual(resp.status_code, status.HTTP_200_OK)
         self.assertIsNotNone(data)
 
+    def test_get_related_products(self):
+        """ Get related products by id tests"""
+        # Test for valid id
+        recommendation = self._create_recommendations(count=20, by_status=True)
+    
+        resp = self.app.get\
+            ("/recommendations/" + str(recommendation[0][0].id))
+
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        self.assertEqual(isinstance(resp, object), True, "Received incorrect recommendation")
+
+        if recommendation[0][0].typeid == 1: 
+            self.assertEqual(resp.get_json()[0]["ids"][0], recommendation[0][0].rel_id, "Received incorrect records")
+        elif recommendation[0][0].typeid == 2: 
+            self.assertEqual(resp.get_json()[1]["ids"][0], recommendation[0][0].rel_id, "Received incorrect records")
+        else: 
+            self.assertEqual(resp.get_json()[2]["ids"][0], recommendation[0][0].rel_id, "Received incorrect records")
+
+        num_records = len(resp.get_json()[0]["ids"]) + len(resp.get_json()[1]["ids"]) + len(resp.get_json()[2]["ids"])
+
+        self.assertEqual(num_records, 1, "Received incorrect records")
```

Comment:
what if num_records = 2, but the number of records with id=recommendation[0][0].id = 10? Can you do something like this at line 95
```
num_expected_records = 0
for rec in recommendation:
    if rec[0][0].id == recommendation[0][0].id:
        num_expected_records += 1
self.assertEqual(num_records, num_expected_records, "Received incorrect number of records")
```

Reply:
I saw in RecommendationFactory, a new id is generated every time the class initialized:  `id = FuzzyChoice(choices=range(1, 10001))`. So when the code calls `self._create_recommendations(count=20, by_status=True)`. it creates 20 records with different ids. Therefore when you use the API to query a given id, only 1 matching record can be found among one of the 3 relationship_types.

Reply:
Oh okay makes sense